### DISCUSSION TITLE
MNI Registration changes real volumes

### DISCUSSION THREAD

**Discussion by qwang0225** [Project-MONAI/MONAI#7361] [Q&A]
Hi, I have been exploring MRI registration recently. **After registering a Nifti format MRI image on MNI152 template, I found out that real total volumes of the image increased**, which is not supposed to happen to my understanding. The real volume = total number of voxels * real volume of each voxel. Please see the code below. 

```
fixed_image = ants.image_read(template_path)
moving_image = ants.image_read(data_path)
transform = ants.registration(fixed_image,moving_image,'Affine')

reg3t = ants.apply_transforms(fixed_image,moving_image,transform['fwdtransforms'][0])
ants.image_write(reg3t, output_path)

mov_nii = nib.load(data_path)
out_nii = nib.load(output_path)


print(mov_nii.shape, out_nii.shape)
print(mov_nii.header.get_zooms(), out_nii.header.get_zooms())
print(np.prod(mov_nii.header.get_zooms()) * np.prod(mov_nii.shape), np.prod(out_nii.header.get_zooms()) * np.prod(out_nii.shape))
```
Here is the output. 
```
(176, 198, 160)        (181, 217, 181)
(0.937, 0.937, 0.937)     (1.0, 1.0, 1.0)
4586871.7        7109137.0
```
So clearly registration changed the image spacing, so the shape of image and unit voxel changed, but I would expect the total volume remain the same. The reason I'm asking is that I try to calculate a segmented volume after feeding the registered image into a binary segmentation model, by multiplying the number of positive voxels and the volume of unit pixel based on output mask.  I want to make sure that this way I will get the real volume of the segmentation. 

Can any of you help out? Where did I miss? Thanks.

**Comment by KumoLiu** [comment]
Hi @ qwang0225, I suggest you ask the question in the ants repo.
https://discourse.slicer.org/t/the-difference-of-methods-in-ants-registration/25236
https://github.com/ANTsX/ANTs/wiki/Anatomy-of-an-antsRegistration-call

BTW, you can refer to 
https://github.com/Project-MONAI/tutorials/blob/main/3d_registration/learn2reg_oasis_unpaired_brain_mr.ipynb
https://github.com/Project-MONAI/tutorials/blob/main/3d_registration/learn2reg_nlst_paired_lung_ct.ipynb

Hope it helps, thanks!
